### PR TITLE
NTFS file system for MAIN partition

### PR DESCRIPTION
**Pull request by tiopex**
Use NTFS file system for MAIN partition.

### CODE REVIEW

File: board/miyoo/main/apps/fsck/fsck.sh [hunk 1]
```diff
@@ -13,18 +13,6 @@ echo -e "\e[32mChecking FAT32 (BOOT) partition...\e[0m"
 fsck.vfat -a /dev/mmcblk0p1
 fi
 
-if dmesg | grep "mmcblk0p4"; then
```
Comment (Apaczer) [inline board/miyoo/main/apps/fsck/fsck.sh:16]: No way of repairing mounted NTFS partition (I see theres ntfsfix in main while mounted)? Then I think no need to keep this app and move `fsck.vfat -a /dev/mmcblk0p1` to /etc/main script

File: board/miyoo/rootfs/etc/main [hunk 2]
```diff
@@ -76,14 +76,17 @@ elif test -r "${HOME}/firstboot.completed"; then
...
+##Check if ntfs is flagged as "dirty", and if so unmount, repair, remount
 	if dmesg | grep "mmcblk0p4" > /dev/null;  then
 		echo -e "\e[31mUnclean shutdown detected.\e[0m"
-		echo -e "\e[32mChecking FAT32 partition...\e[0m"
-		umount /dev/mmcblk0p4
-		fsck.vfat -y /dev/mmcblk0p4 > /dev/null; 
-		mount /dev/mmcblk0p4 "${HOME}" -t vfat -o rw,sync,utf8
+		echo -e "\e[32mChecking NTFS partition...\e[0m"
+		ntfsfix -d /dev/mmcblk0p4 > /dev/null; 
+		mount /dev/mmcblk0p4 "${HOME}" -t ntfs3
```
Comment (Apaczer) [inline board/miyoo/rootfs/etc/main:85]: why mount if there's no umount?

File: board/miyoo/boot/firstboot [hunk 3]
```diff
@@ -514,25 +514,14 @@ SD_MAX_UNALL_SIZE_IN_GiB=$(echo "scale=1; ${SD_MAX_UNALL_SIZE} * 512 / 1024 / 10
...
-# Expand_MAIN_Partition-prompt for fatresize the p4 partition if requested
+# Expand_MAIN_Partition-prompt for ntfsresize the p4 partition if requested
 RESIZE_COMMENCED=false
 if (dialog --clear --stdout --ok-label YES --cancel-label NO --title "   AUTO-RESIZE" \
    --colors --pause "\n\n       \ZuExpand MAIN partition?\Zn\n\n   Select an option & press START
-   \n\n             \ZbWARNING:\Zn\n       After 10s auto-resize\n   of FAT32 partition will begin." 15 60 10); then
+   \n\n             \ZbWARNING:\Zn\n       After 10s auto-resize\n   of NTFS partition will begin." 15 60 10); then
 	clear
 	RESIZE_ABORT=false
-	if test "$SD_SIZE_IN_GiB" -gt 128; then
```
Comment (Apaczer) [inline board/miyoo/boot/firstboot:524]: did you test it on bigger SD cards?

**Review by Apaczer — COMMENTED** [review]
Looks promising, the resizing was almost instantaneous for 16GB. There are still fs corruption but repairs are also fast, let's hope there will be less fatal data loss.

**Comment by tiopex** [comment]
> Looks promising, the resizing was almost instantaneous for 16GB. There are still fs corruption but repairs are also fast, let's hope there will be less fatal data loss.

Unfortunately, after tests, ntfsfix cannot repair all corruptions and only marks the partition as clean. This is a straight road to fatal data loss. So I will give up with NTFS, and try BTRFS.

File: board/miyoo/main/apps/fsck/fsck.sh (hunk 1 above)
Comment (tiopex) [reply board/miyoo/main/apps/fsck/fsck.sh:16]: it is not mounted. When the partition is dirty dmesg informs about it, and refuses to mount

File: board/miyoo/rootfs/etc/main (hunk 2 above)
Comment (tiopex) [reply board/miyoo/rootfs/etc/main:85]: as above

File: board/miyoo/boot/firstboot (hunk 3 above)
Comment (tiopex) [reply board/miyoo/boot/firstboot:524]: only 64gb

**Pull request closed without merging**